File: app/src/main/java/com/example/pcmall/service/CategoryService.java
package com.example.pcmall.service;

import com.example.pcmall.model.Category;
import com.example.pcmall.model.response.ResponseResult;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.POST;

public interface CategoryService {
    @POST("searchCategoryList")
    Observable<ResponseResult<List<Category>>> searchCategoryList();
}

File: core/network/src/main/java/com/example/pcmallcompose/core/network/service/CategoryService.kt
package com.example.pcmallcompose.core.network.service

import com.example.pcmallcompose.core.model.Category
import com.example.pcmallcompose.core.model.response.ResponseResult
import retrofit2.http.POST

interface CategoryService {
    @POST("searchCategoryList")
    suspend fun searchCategoryList(): ResponseResult<List<Category>>
}